src/pymor/vectorarrays/block.py

```diff
@@ -178,8 +189,9 @@ def __eq__(self, other):
     def dim(self):
         return sum(subspace.dim for subspace in self.subspaces)
 
-    def zeros(self, count=1, reserve=0):
-        return BlockVectorArray([subspace.zeros(count=count, reserve=reserve) for subspace in self.subspaces], self)
+    def zeros(self, count=1, reserve=0, dtype=None):
+        return BlockVectorArray([subspace.zeros(count=count, reserve=reserve, dtype=None)
+                                 for subspace in self.subspaces], self)
```

Comment:
I guess we agreed on your option 2 in #333, but then limited `dtype` in `zeros` to `NumpyVectorArray` in this thread. Then I realized that it would make sense that `BlockVectorArray.zeros` also has `dtype` if it is created from `NumpyVectorArray`s. Now I see that this code would not work for arbitrary `BlockVectorArray`.
I'm not sure what to do here. Should I add a check `isinstance(subspace, NumpyVectorSpace)`, or add `dtype` everywhere, or remove it from everywhere?